### ISSUE TITLE
Reorder dfocc:dfmp2 Contractions to Remove Bottleneck

### ISSUE THREAD

**Issue by JonathonMisiewicz** [psi4/psi4#1726]
MP2 only requires the `(ia|ia)` block of two-electron integrals. In density-fitting, this means only the `(P|Q)^-1/2 (Q|ia)` block of three-center ERIs is needed.

Rob's `dfmp2` code transforms the `|μν)` integrals to `|ia)` first and then multiplies by `(P|Q)^-1/2`. Ugur's code multiplies by `(P|Q)^-1/2` first and then transforms the `|μν)` quantities to `|ia)`. I notice that at least for RHF (UHF not tested) Rob's code is much faster than Ugur's for this integral transformation step. Which isn't too surprising. The integral transformations have the same scaling, but Rob's `(P|Q)^-1/2` step is _O(aux^2 mo^2)_, while Ugur's is _O(aux^2 occ vir)_. Ugur's procedure makes sense in the general case where you also need `|ii)` and `|aa)` quantities, but not for MP2.

After #1390 goes in, @bozkaya, would it make sense to have custom logic for DFMP2 to change the order of contractions? I think we would see significant speedups.

**Comment by susilehtola** [comment]
Doesn’t the orbital gradient need the extra blocks?

**Comment by JonathonMisiewicz** [comment]
Yes, but I'm referring to canonical-orbital MP2, not OMP2. Although Psi defaults to Rob's implementation, `dfocc` has an implementation of canonical-orbital DF-MP2 as well.

**Comment by bozkaya** [comment]
I have known this issue for a long time. However, my integral transform is written for CC methods, not specifically for DF-MP2. Since I use the same code for all methods I have no plan for a DF-MP2 specific transformation in the near future. We can use Rob's code as the default df-mp2 code. @JonathonMisiewicz @susilehtola

**Comment by JonathonMisiewicz** [comment]
Closed as "won't fix," in that case.
